src/components/routes/CalendarItemDisplay.vue

```diff
@@ -0,0 +1,241 @@
+ * props. Each shows arrow pointing in a different direction.
+ *
+ * @props
+ * - `day` (RouteCalendarDay, default - null): The object representing the associated route.
+ *   It should be of type `RouteCalendarDay`.
+ * - `toWork` (boolean, optional): Whether the route is to work.
+ * - `fromWork` (boolean, optional): Whether the route is from work.
+ * - `active` (boolean, optional): Whether the route is active (currently
+ *   being edited).
+ *
+ * @events
+ * - `item-click`: Emitted when the item is clicked.
+ *   payload:
+ *     - `direction` (TransportDirection): The direction of the route.
+ *     - `timestamp` (Timestamp): The timestamp of the route.
+ *
+ * @example
+ * <calendar-item
+ *  direction="toWork"
+ *  :day="routeCalendarDay"
+ *  :timestamp="timestamp"
+ * />
+ *
+ * @see [Figma Design](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=8279-39418&t=baRxKn3ERzsqCVyl-1)
+ */
+
+// libraries
+import { computed, defineComponent } from 'vue';
+
+// composables
+import { useRoutes } from 'src/composables/useRoutes';
+
+// types
+import type { Timestamp } from '@quasar/quasar-ui-qcalendar';
+import type {
+  RouteCalendarDay,
+  RouteItem,
+  TransportDirection,
+} from '../types/Route';
+
+export default defineComponent({
+  name: 'CalendarItemDisplay',
+  props: {
+    day: {
+      type: Object as () => RouteCalendarDay | null,
+      default: null,
+    },
+    direction: {
+      type: String as () => TransportDirection,
+      required: true,
+    },
+    timestamp: {
+      type: Object as () => Timestamp,
+      required: true,
+    },
+    active: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  emits: ['item-click'],
+  setup(props, { emit }) {
+    function onClick(): void {
+      emit('item-click', {
+        timestamp: props.timestamp,
+        direction: props.direction,
+      });
+    }
+
+    const route = computed((): RouteItem | null => {
+      if (!props.day) return null;
+      return props.day[props.direction] ? props.day[props.direction] : null;
+    });
+
+    const { getRouteIcon } = useRoutes();
+
+    return {
+      route,
+      getRouteIcon,
+      onClick,
+    };
+  },
+});
+</script>
+
+<template>
+  <div class="q-my-sm" data-cy="calendar-item-display">
+    <q-item
+      :active="active"
+      dense
+      clickable
+      v-ripple
+      class="relative-position flex justify-center items-center text-center gap-8"
+      @click.prevent="onClick()"
+      data-cy="calendar-item-display-item"
+    >
+      <!-- Background: To work -->
+      <template v-if="direction === 'toWork'">
+        <!-- Icon: To work - active -->
+        <q-icon
+          v-if="active"
+          color="primary"
+          class="full-width full-height absolute-full"
+          data-cy="calendar-item-icon-towork-active"
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 135 38">
```

Comment:
Should be standalone SVG image.

Reply:
Refactored using [public svg with svg:use](https://quasar.dev/vue-components/icon#svg-use-way)
Will be used for new custom SVGs.